File: RogueTraderSystemGenerator/Nodes/GravityRiptideNode.cs
using System;
using System.Runtime.Serialization;
using System.Windows.Documents;

namespace RogueTraderSystemGenerator.Nodes
{
    [DataContract]
    class GravityRiptideNode : NodeBase
    {
        public GravityRiptideNode() 
        {
            _nodeName = "Gravity Riptide";
        }

        public override void GenerateFlowDocument()
        {
            _flowDocument = new FlowDocument();
            DocBuilder.AddHeader(ref _flowDocument, NodeName, 3);
            if (!String.IsNullOrEmpty(Description))
                DocBuilder.AddContentLine(ref _flowDocument, "Description", new DocContentItem(Description));
            DocBuilder.AddContentLine(ref _flowDocument, "", new DocContentItem("Gravity Riptides follow the rules for Gravity Tides on page 227 of the Rogue Trader Core Rulebook. ", 15, "Gravity Riptide"));
        }

        public override void Generate()
        {
        }
    }
}

File: electron-app/js/nodes/gravityRiptideNode.js
// Create the remaining node files

// GravityRiptideNode.js
class GravityRiptideNode extends NodeBase {
    constructor(id = null) {
        super(NodeTypes.GravityRiptide, id);
        this.nodeName = 'Gravity Riptide';
        this.fontWeight = 'bold';
        this.headerLevel = 3; // H3: Feature/hazard node
    }

    generate() {
        // No random attributes in C# version; keep minimal
        super.generate();
        this.updateDescription();
    }

    updateDescription() {
        // No duplicate header - base class will add H3
        // Parity with C# (static rules reference line)
        this.description = `<p>Gravity Riptides follow the rules for Gravity Tides on page 227 of the Rogue Trader Core Rulebook.</p>`;
    }

    static fromJSON(data) {
        const node = new GravityRiptideNode(data.id);
        Object.assign(node, data);
        node.updateDescription();
        return node;
    }
}

window.GravityRiptideNode = GravityRiptideNode;